File: fizzBuzz.js
var isFizzNumber = function(number) {
	if(number % 3 === 0)
		console.log('Fizz');
}

var isBuzzNumber = function(number) {
	if(number % 5 === 0)
		console.log('Buzz');
}

var isFizzBuzzNumber = function(number) {
	if(number % 5 === 0 && number % 3 === 0)
		console.log('FizzBuzz');
}

for (var i = 0; i <= 33; i++) {
	if(!isBuzzNumber && !isFizzNumber && !isFizzBuzzNumber)
		console.log(i);
};

File: FizzBuzz/ruby/lib/fizzbuzz.rb

class FizzBuzz
  

  def initialize
    @is_durex = ->(number) { number == 69 }

    @is_fizzbuzz = ->(number) { number % 3 == 0 && number % 5 == 0 }
    @is_fizz = ->(number) { number % 3 == 0 }
    @is_buzz = ->(number) { number % 5 == 0 }

    @lambdas =  [ 
      [@is_durex, "Durex"],
      [@is_fizzbuzz, "FizzBuzz"],
      [@is_fizz, "Fizz"],
      [@is_buzz, "Buzz"]
       ]              
  end

  def check_number(number)
      @lambdas.each { |pair| if pair[0].call(number) then return pair[1] end}
       
    end
end

